src/main/java/org/jabref/logic/journals/PredatoryJournalLoader.java

```diff
@@ -0,0 +1,127 @@
+        new PJSource("https://raw.githubusercontent.com/stop-predatory-journals/stop-predatory-journals.github.io/master/_data/journals.csv",
+                    null),
+        /*
+        new PJSource("https://raw.githubusercontent.com/stop-predatory-journals/stop-predatory-journals.github.io/master/_data/hijacked.csv",
+                    null,
+                    null,
+                    "journal", "journalname", "bookname"),
+        */
+        new PJSource("https://raw.githubusercontent.com/stop-predatory-journals/stop-predatory-journals.github.io/master/_data/publishers.csv",
+                    null),
+        new PJSource("https://beallslist.net/",
+                    "<li>.*?</li>"),
+        new PJSource("https://beallslist.net/standalone-journals/",
+                    "<li>.*?</li>"),
+        new PJSource("https://beallslist.net/hijacked-journals/",
+                    "<tr>.*?</tr>")
+    );
+
+    private static final Logger                 LOGGER              = LoggerFactory.getLogger(PredatoryJournalLoader.class);
+    private static HttpClient                   client;
+    private static List<String>                 linkElements;
+    private static PredatoryJournalRepository   repository;
+
+    public PredatoryJournalLoader() {
+        this.client             = HttpClient.newHttpClient();
+        this.linkElements       = new ArrayList<>();
+    }
+
+    public static PredatoryJournalRepository loadRepository() {
+        // Initialize in-memory repository
+        repository = new PredatoryJournalRepository();
+
+        // Update from external sources
+        update();
+
+        return repository;
+    }
+
+    private static void update() {
+        PREDATORY_SOURCES   .forEach(PredatoryJournalLoader::crawl);            // populates linkElements (and predatoryJournals if CSV)
+        linkElements        .forEach(PredatoryJournalLoader::clean);            // adds cleaned HTML to predatoryJournals
+
+        LOGGER.info("UPDATED PREDATORY JOURNAL LIST");
+    }
+
+    private static void crawl(PJSource source) {
+        var uri     = URI.create(source.URL);
+        var request = HttpRequest.newBuilder().uri(uri).build();
+
+        try {
+            HttpResponse<String> response = client.send(request, BodyHandlers.ofString());
+
+            if (response.statusCode() != 200)       { LOGGER.warn("BAD RESPONSE"); }
+            else if (source.URL.contains(".csv"))   { handleCSV(response.body()); }
+            else                                    { handleHTML(source.ELEMENT_REGEX, response.body()); }
+        }
+        catch (IOException ex)          { logException(ex); }
+        catch (InterruptedException ex) { logException(ex); }
+    }
+
+    private static void handleCSV(String body) {
+        var csvSplit = Pattern.compile("(\"[^\"]*\"|[^,]+)");
+
+        for (String line : body.split("\n")) {                                  // TODO: skip header
```

Comment:
You can use the CSV library we already have included
 https://github.com/JabRef/jabref/blob/9d3bcf3d98aa8eedfdf0ca2ec9790baa31291f83/src/main/java/org/jabref/logic/journals/AbbreviationParser.java#L57

Reply:
Ah good, I didn't see that there was one included. Will switch to that.